penguin/integrations/mcp/server_tools/ituv.py

```
@@ -0,0 +1,619 @@
+        ],
+    }
+
+
+def _safe_project_dag_stats(project_manager: Any, project_id: str) -> dict[str, Any]:
+    try:
+        return project_manager.get_dag_stats(project_id)
+    except Exception as exc:
+        return {"error": "dag_stats_failed", "message": str(exc)}
+
+
+def _task_readiness(project_manager: Any, task: Any) -> dict[str, Any]:
```

Ruff (0.15.12)

[warning] Dynamically typed expressions (typing.Any) are disallowed in `project_manager`

(ANN401)